src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/BlobProcessor.java

```diff
@@ -45,16 +66,26 @@ public void process(String blobName, String containerName) {
             byte[] rawBlob = tryToDownloadBlob(blobClient);
 
             dispatcher.dispatch(blobName, rawBlob, containerName);
+            UUID envelopeId = envelopeRepository.insert(createNewEnvelope(
+                blobName,
+                containerName,
+                blobClient.getProperties().getCreationTime().toInstant()
+            ));
 
-            logger.info("Finished processing {} from {} container", blobName, containerName);
+            logger.info(
+                "Finished processing {} from {} container. New envelope ID: {}",
+                blobName,
+                containerName,
+                envelopeId
+            );
         } catch (Exception exception) {
             logger.error("Error occurred while processing {} from {}", blobName, containerName, exception);
```

Comment:
If we failed to acquire lease, we should just log it as info and continue with other blobs.

Reply:
After all this usage we actually will only get (as far as I have expected) `BlobStorageException` so we need to distinguish between different steps. Perhaps need to extract each step to isolated/different method call so to be able to catch all exceptions individually

Reply:
Good point problem acquiring lease.
Acquiring lease and handling exception existed before this PR, so I think it would be better to address this in a separate PR.